crates/code_generation/src/cyclers.rs

```diff
@@ -622,6 +729,96 @@ fn generate_cross_inputs_recording(
+                }
+            }
+            Field::HistoricInput { path, data_type, .. } => {
+                let name = path_to_extraction_variable_name("own", &path, "historic_input");
+                quote! {
+                    #[allow(non_snake_case)]
+                    let #name: std::collections::BTreeMap<std::time::SystemTime, #data_type> = bincode::deserialize_from(&mut recording_frame).wrap_err(#error_message)?;
+                }
+            }
+            Field::Input {
+                cycler_instance: Some(cycler_instance),
+                path,
+                data_type,
+                ..
+            } => {
+                let name = path_to_extraction_variable_name(&cycler_instance, &path, "input");
+                quote! {
+                    #[allow(non_snake_case)]
+                    let #name: #data_type = bincode::deserialize_from(&mut recording_frame).wrap_err(#error_message)?;
+                }
+            }
+            Field::PerceptionInput { cycler_instance, path, data_type, .. } => {
+                let name = path_to_extraction_variable_name(&cycler_instance, &path, "perception_input");
+                quote! {
+                    #[allow(non_snake_case)]
+                    let #name: [std::collections::BTreeMap<std::time::SystemTime, Vec<#data_type>>; 2] = bincode::deserialize_from(&mut recording_frame).wrap_err(#error_message)?;
+                }
+            }
+            Field::RequiredInput {
+                cycler_instance: Some(cycler_instance),
+                path,
+                data_type: Type::Path(TypePath {
+                    path: SynPath { segments, .. },
+                    ..
+                }),
+                ..
+            } if !segments.is_empty() && segments.last().unwrap().ident == "Option" => {
+                let name = path_to_extraction_variable_name(&cycler_instance, &path, "required_input");
+                let data_type = match &segments.last().unwrap().arguments {
+                    PathArguments::AngleBracketed(AngleBracketedGenericArguments {
+                        args, ..
+                    }) if args.len() == 1 => match args.first().unwrap() {
+                        GenericArgument::Type(nested_data_type) => nested_data_type,
+                        _ => panic!("unexpected generic argument, expected type argument in data type"),
+                    },
+                    _ => panic!("expected exactly one generic type argument in data type"),
+                };
+                quote! {
+                    #[allow(non_snake_case)]
+                    let #name: #data_type = bincode::deserialize_from(&mut recording_frame).wrap_err(#error_message)?;
+                }
+            }
+            _ => panic!("unexpected field {field:?}"),
+        }
+    }).collect::<Vec<_>>();
+
+    if extractions.is_empty() {
+        return Default::default();
+    }
```

Comment:
What is the point of this extra check?
Doesn't the quote below return an empty token stream as well when there are no elements in `extractions`?

Reply:
There is no point in this extra check. This was a copy :spaghetti: mistake because for recordings we need this check. :wink: